### PR TITLE
feat: add event tracking xlsx data export for octoware

### PR DESCRIPTION
**Pull request by mad-nuts**
refactor: rework data-export code to separate data specific payload and general business logic
refactor: replace custom export dialog with vuetify dialog
refactor: move data export logic to dedicated components / services

### CODE REVIEW

**Review by jekutzsche — DISMISSED** [review]
When looking at the file `iris-client-fe/src/utils/data-export/data-export.ts` it occurred to me that there is quite a lot in it. The export and the sanitization could also be seen as two responsibilities and separate them.

**Review by jekutzsche — DISMISSED** [review]

File: iris-client-fe/src/utils/data-export/sanitization.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+const sanitizeLineBreaks = (field: string): string => {
+  // Replace newlines and other line breaks with a space
+  // Strip whitespace at the beginning and end
+  const regex_linebreaks = / *(\r?\n|\r) */g; // Recognizes /r (CR), /n (LF) and /r/n (CRLF) including surrounding spaces
+  return field.replace(regex_linebreaks, " ").trim();
+};
+
+const sanitizeWhitelist = (field: string): string => {
+  // Remove everything not whitelisted (this restriction may be relaxed at some point)
+  const regex_whitelist = /[^a-zA-Z0-9äüöÄÜÖß(): \-@+.;,]+/g; // Matches everything *not* in the group (the whitelist)
```
[the remark this answers is not in the document]
Comment (bkastl) [reply iris-client-fe/src/utils/data-export/sanitization.ts:15]: moved to separate issue

**Review by bkastl — APPROVED** [review]
Export to excel okay, new findings to follow in separate issue

**Review by bkastl — APPROVED** [review]

**Pull request merged**